### PR TITLE
Bug 1202939 - Addons are sorted by name in alphabetical order, r=gasolin

### PR DESCRIPTION
**Pull request by scottwu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9c26422406b10fece90cdf97c88f0573746b2423

File: apps/settings/js/panels/addons/addons_list.js [hunk 1]
```diff
@@ -34,13 +35,21 @@ define(function(require) {
...
+
     _update: function() {
       return this._filter ?
         Promise.all(this._addonManager.addons.array.map(addon =>
           this._addonManager.addonAffectsApp(addon, this._filter).then(
             affects => affects ? addon : undefined))).then(
-              addons => this._addons.reset(addons.filter(addon => addon))) :
-        Promise.resolve(this._addons.reset(this._addonManager.addons.array));
+              addons => this._addons.reset(this._sortByName(addons.filter(
+                addon => addon)))) :
```
Comment (gasolin) [inline apps/settings/js/panels/addons/addons_list.js]: It looks more eye friendly to break line at the same end `this._sortByName(`
addons.filter(addon => addon)))) : :)

**Comment by gasolin** [comment]
The commit comment could be more accurate, such as `sort Addons by name in alphabetical order`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fc9197b3ef7f834eb8d5a4101ace7fe5978649a7

File: apps/settings/js/panels/addons/addons_list.js [hunk 2]
```diff
@@ -34,13 +35,22 @@ define(function(require) {
...
+
     _update: function() {
       return this._filter ?
         Promise.all(this._addonManager.addons.array.map(addon =>
           this._addonManager.addonAffectsApp(addon, this._filter).then(
             affects => affects ? addon : undefined))).then(
-              addons => this._addons.reset(addons.filter(addon => addon))) :
-        Promise.resolve(this._addons.reset(this._addonManager.addons.array));
+              addons => this._addons.reset(
+                this._sortByName(
```
Comment (gasolin) [inline apps/settings/js/panels/addons/addons_list.js]: I mean it's better looks the same with below syntax `this._addons.reset(this._sortByName(`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b0df4a175db36bdd1a787d5b1b80c9fc8b404328

File: apps/settings/js/panels/addons/addons_list.js (hunk 2 above)
Comment (scottwu) [inline apps/settings/js/panels/addons/addons_list.js]: My mistake :P 
It should be fine now.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=42d413f1b1f37962ded6dcf5d536b7577e97ce1e

**Pull request merged**